src/plugins/plugin-http.ts

```diff
@@ -137,20 +141,37 @@ function patchHttp(http, api) {
 
   if (semver.satisfies(process.version, '>=8.0.0')) {
     // http.get in Node 8 calls the private copy of request rather than the one
-    // we have patched on module.export. We need to patch get as well. Luckily,
-    // the request patch we have does work for get as well.
-    shimmer.wrap(http, 'get', function getWrap(get) {
-      return makeRequestTrace(get, api);
+    // we have patched on module.export, so patch get as well.
+    shimmer.wrap(http, 'get', function getWrap() {
+      // Re-implement http.get. This needs to be done (instead of using
+      // makeRequestTrace to patch it) because we need to set the trace
+      // context header before the returned ClientRequest is ended.
+      // The Node.js docs state that the only differences between request and
+      // get are that (1) get defaults to the HTTP GET method and (2) the
+      // returned request object is ended immediately.
+      // The former is already true (at least in supported Node versions up to
+      // v9), so we simply follow the latter.
+      // Ref: https://nodejs.org/dist/latest/docs/api/http.html#http_http_get_options_callback
+      return function makeGetTrace() {
```

Comment:
Any reason to not explicitly pass the two arguments here? Last I checked, apply() has a performance hit over an explicit function call

Reply:
`apply` seemed incrementally less intrusive just because it doesn't modify the implicit argument -- I didn't know it incurs a performance hit over the explicit call. Since `this` is irrelevant I'll change this.